submissions/nadsatt/friends-app/modules/filter-group.js

```diff
@@ -0,0 +1,93 @@
+import * as searchFiltersModule from './search-filter-items.js';
+import * as sortFiltersModule from './sort-filter-items.js';
+import { FilterList } from './filter-list.js';
+import { IconList } from './icon-list.js';
+
+export class FilterGroup {
+    constructor(userService, pageLinkList){
+        this.defineElement(userService, pageLinkList);
+        this.defineElementMethods();
+
+        return this.element;
+    }
+
+    defineElement(userService, pageLinkList){
+        const categories = ['sort', 'search'];
+        const filtersModules = [sortFiltersModule,
+                                searchFiltersModule];
+
+        [this[`${categories[0]}FilterClasses`],
+         this[`${categories[1]}FilterClasses`]] = filtersModules.map(filtersModule =>
+            Object.values(filtersModule)
+        );
+
+        this.element = document.createElement('div');
+        this.element.classList.add('filter-group');
+        this.element.userService = userService;
+        this.element.pageLinkList = pageLinkList;
+
+        [this.element[`${categories[0]}Filters`],
+         this.element[`${categories[1]}Filters`]] = categories.map(category =>
+            this[`${category}FilterClasses`].map(FilterClass =>
+                new FilterClass(category, userService))
+        );
+
+        this.element.filters = categories.reduce((res, category) => {
+            res.push(...this.element[`${category}Filters`]);
+            return res;
+        }, []);
+
+        [this.element[`${categories[0]}FilterList`],
+         this.element[`${categories[1]}FilterList`]] = categories.map(category =>
+            new FilterList(this.element[`${category}Filters`], category)
+        );
+
+        [this.element[`${categories[0]}IconList`],
+         this.element[`${categories[1]}IconList`]] = categories.map(category =>
+            new IconList(this.element[`${category}FilterList`], category)
+        );
+
+        this.element.append(...categories.map(category => {
```

Comment:
Please move callback into the function with descriptive name

Reply:
@zonzujiro  i found a solution for [this](https://github.com/kottans/frontend-2021-homeworks/pull/338#issuecomment-753361078) problem, so to save your time and avoid double re-review you can answer and view the code after i will commit changes.

Reply:
So you have placed element into separate property of instance?

I meant, that you should operate "only" with instances. And then re-render result.